src/psyclone/transformations.py

```diff
@@ -1361,3 +1363,83 @@ def apply(self, loop, depth=None):
         loop.update_halo_exchanges()
 
         return schedule, keep
+
+
+class LoopSwapTrans(Transformation):
+    ''' Provides a loop-swap transformation.
+        For example:
+
+        >>> from parse import parse
+        >>> from psyGen import PSyFactory
+        >>> ast,invokeInfo=parse("dynamo.F90")
+        >>> psy=PSyFactory("dynamo0.1").create(invokeInfo)
+        >>> schedule=psy.invokes.get('invoke_v3_kernel_type').schedule
+        >>> schedule.view()
+        >>>
+        >>> from transformations import LoopSwapTrans
+        >>> swap=LoopSwapTrans()
+        >>> new_schedule,memento=swap.apply(schedule.children[0],
+                                             schedule.children[1])
+        >>> new_schedule.view()
+    '''
+
+    def __str__(self):
+        return "Exchange the order of two nested loops: inner becomes " + \
+               "outer and vice versa"
+    @property
+    def name(selfself):
```

Comment:
Is `selfself` what you meant? (Update: I guess not because pylint doesn't like it)